torch/_subclasses/fake_tensor.py

```diff
@@ -560,9 +560,13 @@ def _memo_epoch(self, obj: FakeTensor) -> str:
 
     def __get__(
         self, obj: FakeTensor, objtype: Optional[Type[FakeTensor]] = None
-    ) -> Optional[torch.SymInt]:
+    ) -> Optional[Union[torch.SymInt, torch.SymFloat]]:
         if (r := getattr(obj, self._memo(obj))) is None:
             return None
+
+        if isinstance(r, torch.SymFloat):
+            return r
```

Comment:
Memo is unconditionally ignored for SymFloat memos

Reply:
A tighter condition probably involves testing if something has any unbacked things in it, because if there aren't any (and with floats you wouldn't expect there to be e.g., zf0), then that's when it's OK to preserve the memo (because it won't renumber)